### PR TITLE
adjust dev_cd to trigger on new release

### PR DESCRIPTION
**Pull request by dontseyit**
[GDEV-1827](https://jira.verbis.dkfz.de/browse/GDEV-1827)

### CODE REVIEW

File: .github/workflows/dev_cd.yaml [hunk 1]
```diff
@@ -1,49 +1,60 @@
 name: Dev CD
```
Comment (KerstenBreuer) [inline .github/workflows/dev_cd.yaml:1]: Also rename the file to cd.yaml.

```suggestion
name: CD
```

File: .github/workflows/dev_cd.yaml [hunk 2]
```diff
@@ -1,49 +1,60 @@
...
       # export to be used in other jobs
-      version: ${{ steps.get_commit_version.outputs.version }}
-      branch: ${{ steps.get_commit_version.outputs.branch }}
+      version: ${{ steps.get_version_tag.outputs.version }}
     steps:
       - uses: actions/checkout@v3
         name: Check out code
+
+      - uses: actions/setup-python@v4
+        name: Set up Python 3.x
```
Comment (KerstenBreuer) [inline .github/workflows/dev_cd.yaml:19]:
```suggestion
        name: Set up Python 3.9
```

File: .github/workflows/dev_cd.yaml [hunk 3]
```diff
@@ -1,49 +1,60 @@
...
-        name: get commit version
+      - id: verify_semantic_tag_format
+        name: Verify tag format
+        # format must be compatible with semantic versioning
         run: |
-          BRANCH="$(git rev-parse --abbrev-ref HEAD)"
-          if git describe --tags
-          then
-            VERSION="$(git describe --tags)-${BRANCH}"
+          SEMVER_REGEX="^(0|[1-9][0-9]*)\.(0|[1-9][0-9]*)\.(0|[1-9][0-9]*)(\-[0-9A-Za-z-]+(\.[0-9A-Za-z-]+)*)?(\+[0-9A-Za-z-]+(\.[0-9A-Za-z-]+)*)?$"
```
Comment (KerstenBreuer) [inline .github/workflows/dev_cd.yaml:35]: The regex suggested on https://semver.org/ is the following:

```suggestion
          SEMVER_REGEX="^(0|[1-9]\d*)\.(0|[1-9]\d*)\.(0|[1-9]\d*)(?:-((?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*)(?:\.(?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*))*))?(?:\+([0-9a-zA-Z-]+(?:\.[0-9a-zA-Z-]+)*))?$"
```

**Review by KerstenBreuer — CHANGES_REQUESTED** [review]
Thanks a lot. Looks great. For the automated update of repos relying on the template, could you add the dev_cd.yaml to the `.deprecated_files` and remove it from `.mandatory_files` and `.static_files` if needed. Moreover, could you add the cd.yaml to `.static_files`?

Have you checked this action on a dummy repo? Would be good to make sure if it works before rolling it out. Testing in the template repo will be difficult.

**Comment by dontseyit** [comment]
@KerstenBreuer 

I updated the branch with your suggestions. You can check again. 

For the tests, I have a personal sandbox repo generated from the template. https://github.com/dontseyit/white-mountains 
I tested there also with my docker hub. 

This is the run recent updates https://github.com/dontseyit/white-mountains/actions/runs/4565363578
and the docker images go to https://hub.docker.com/r/seyit/white-mountains/tags

**Review by KerstenBreuer — APPROVED** [review]
Great thanks.

**Comment by lkuchenb** [comment]
Shall upgrade `docker/login-action@v2.0.0` to `docker/login-action@v2.1.0` while we're at it? The old version seems to trigger deprecation warnings from GitHub, maybe the newer one resolves that.

```
Warning: The `save-state` command is deprecated and will be disabled soon. Please upgrade to using Environment Files. For more information see: https://github.blog/changelog/2022-10-11-github-actions-deprecating-save-state-and-set-output-commands/
```

File: .github/workflows/cd.yaml [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+      - uses: actions/checkout@v3
+        name: Check out code
+
+      - uses: docker/setup-qemu-action@v2.0.0
+        name: Set up QEMU
+
+      - uses: docker/setup-buildx-action@v2.0.0
+        name: Set up Docker Buildx
+
+      - uses: docker/login-action@v2.0.0
```
Comment (lkuchenb) [inline .github/workflows/cd.yaml:68]:
```suggestion
      - uses: docker/login-action@v2.1.0
```

**Review by lkuchenb — CHANGES_REQUESTED** [review]

**Comment by dontseyit** [comment]
These versions are updated in order to fix the save-state warning.

login-action v2.0.0 -> 2.1.0
setup-buildx-action v2.0.0 -> 2.5.0
build-push-action v3.0.0 -> 4.0.0

This is the latest test run with these versions
https://github.com/dontseyit/white-mountains/actions/runs/4574452267/jobs/8076087363

**Review by lkuchenb — APPROVED** [review]
Thanks, looks like all warnings are gone 👍

**Review by KerstenBreuer — APPROVED** [review]

**Pull request merged**